karma.conf.js

```diff
@@ -7,8 +7,19 @@ function depMain(name) {
   return './node_modules/' + name + '/' + main;
 }
 
+//  getDeps :: String -> StrMap String String
+function getDeps(path) {
+  return Object.keys(require(path).dependencies);
+}
+
 //  dependencies :: Array String
-var dependencies = Object.keys(require('./package.json').dependencies);
+var dependencies = getDeps('./package.json')
+      .map(function(x) {
+        return [x, getDeps('./node_modules/' + x + '/package.json')];
+      })
+      .sort(function(a, b) { return a[1].indexOf(b[0]) === -1 ? -1 : 1; })
+      .sort(function(b, a) { return a[1].indexOf(b[0]) === -1 ? 1 : -1; })
```

Comment:
Could you explain why it's necessary to sort twice?

By the way, I prefer `xs.indexOf(x) < 0` to `xs.indexOf(x) === -1`. `-1` is the "bogus" value; it's nice to avoid referencing it directly.